lib/commands/user.js

```diff
@@ -21,207 +21,200 @@
  *
  */
 
-var modulus = require('../modulus'),
-    userConfig = require('../common/api').userConfig,
-    userController = require('../controllers/user'),
-    error = require('../common/error'),
-    util = require('util');
+const Modulus = require('../modulus');
+const UserConfig = require('../common/api').userConfig;
+const UserController = require('../controllers/user');
+const Errors = require('../common/error');
+const Util = require('util');
 
-var user = {};
+var _signupPrompt, user = {};// eslint-disable-line no-underscore-dangle
+var print = Modulus.io.print;
 
-user.authenticate = function(username, password, loginProvided, passwordProvided, cb) {
-  userController.authenticate(
+user.authenticate = function (username, password, loginProvided, passwordProvided, cb) {
+  UserController.authenticate(
     username,
     password,
-    function(err, u) {
-      if(err) {
-        err = error.handleApiError(err, 'LOGIN', cb);
+    function (err, u) {
+      var login = null;
+      var udata = null;
+      if (err) {
+        err = Errors.handleApiError(err, 'LOGIN', cb);
         if (err.length > 0) {
-          modulus.io.error(err);
-
-          if(loginProvided && passwordProvided) {
-            return;
-          } else if(loginProvided) {
-            return user.login(username, null, false, cb);
-          } else if(passwordProvided) {
-            return user.login(null, password, false, cb);
-          } else {
-            return user.login(null, null, false, cb);
+          Modulus.io.error(err);
+
+          if (loginProvided) login = user.login(username, null, false, cb);
```

Comment:
this feels a bit too much like a reactor, and not a lint. Additionally, I think behavior changed here.